pkg/limits/store.go

```diff
@@ -328,24 +362,32 @@ func (s *usageStore) get(i int, tenant string, partition int32, streamHash uint6
 	if !ok {
 		return
 	}
-	streams, ok := partitions[partition]
+	policies, ok := partitions[partition]
 	if !ok {
 		return
 	}
-	stream, ok = streams[streamHash]
-	return
+	// Search across all policies for this stream
+	// Note that in most cases, there will only be one item on this list (empty policy, "").
+	// and at most just a few items. One idea to speed this up would be to keep a cache of tenant-partition-streamHash -> policy for those streams that have a matching policy.
+	for _, streams := range policies {
```

Comment:
I think the issue here is that if a stream is present in two policies at the same time, this is non-deterministic now because iterating maps is non-deterministic in Go.

Reply:
I removed the `Get` method.

In the future if we add this back, I think the stream should be in both the _noPolicy ("")_ bucket and a named policy bucket when we strat configuring those streams to be assigned to a particular policy. In which case I think we could remove the stream form the noPolicy bucket when it is added for the first time.